### DISCUSSION TITLE
Add DOM attribute when next/image is done loading

### DISCUSSION THREAD

**Discussion by trotzig** [vercel/next.js#33871] [Ideas]
### Describe the feature you'd like to request

Hi there! I run a screenshot testing service called happo.io (proudly run on next.js since 2017). Lately I've had to debug a few issues with spurious screenshots coming from images constructed via `next/image`, for [Cypress test runs](https://docs.happo.io/docs/cypress). For screenshot testing to be stable, it's important that we know when images are fully loaded. With `next/image`, it's hard to do that without either: 
- inspecting the style attribute to look for `/filter.*blur/ 
- add a `load` event listener for the image element in user land
- poll images for the `complete` property to be `true`

To be clear, this is mostly a Cypress/e2e testing issue where Happo (and other services, like Percy) take a DOM snapshot of the page/element before sending that DOM snapshot off to a screenshot service. We need to know that the DOM is fully ready. We could add some code that will always wait for images to be downloaded before taking the DOM snapshot, but that would make a number of test suites slower, and we don't want that. We just need the `<img>` element to be populated with enough information so that we can recreate it later. In the case of `next/image`, we need to wait for the DOM to be in the "ready" state -- no blurry background image.  

### Describe the solution you'd like

It would be super helpful if the underlying `img` element got a special attribute/class name when we know it's fully loaded. Something like 

```html
<img data-nimg-ready="true">
```

Or perhaps the image can be in different states: 

```html
<img data-nimg-phase="idle">
<img data-nimg-phase="loading">
<img data-nimg-phase="ready">
```

### Describe alternatives you've considered

This is possible to solve in user land by adding something like 

```js
  cy.get('img').should($imgs =>
    $imgs.map((_i, img) => expect(img.complete).to.be.true)
  )
```

**Comment by danielgent** [comment]
I have the exact same use case (I'm surprised more people don't TBH!)

I made my own wrapper around `next/image` that sets a `data-` attribute once on `onLoadingComplete`

Seems to work great so far for doing snapshots in Cypress
